routes/merged/project/load_balancer.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class Clover
+  branch = lambda do |r|
+    r.get true do
+      load_balancer_list
+    end
+
+    r.on api? do
+      r.on String do |lb_name|
+        r.post true do
+          load_balancer_post(name: lb_name)
+        end
+      end
+    end
+
+    r.on web? do
```

Comment:
This isn't a problem, but the `r.on web?` call is unncessary. With the above `r.on api?` call, you know that everything after that until the end of the block must be a web route and not an api route.

Reply:
If I remember correctly, when we use `r.on`, it only applies to that specific branch. However, `r.is` checks all branches until it finds a match. Am I correct?

So, if it's `r.is api?`, we need to specify this line, right?

Reply:
If we were using `r.is api?` above, then yes, but not for the reason you think.  The only difference between `r.is` and `r.on` is that `r.is` checks that the path has been fully consumed after all matchers are applied.  So with `r.on api?`, all api requests will go to that block, and you know the rest of the requests after `r.on api?` are web requests.  With `r.is api?`, only api requests were the remaining path is empty would go to that block.  After `r.is api?`, you would have all web requests, and also any api requests where the remaining path is not empty.